File: java-to-go-projects/05-port-scanner/rust/src/result.rs
use std::fmt;

pub struct ScanResult {
    pub host: String,
    pub port: u16,
    pub open: bool,
    pub latency_ms: u64,
}

impl fmt::Display for ScanResult {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        fmt::write(
            f,
            format_args!(
                "{}:{} is {} (latency: {} ms)",
                self.host,
                self.port,
                if self.open { "open" } else { "closed" },
                self.latency_ms
            )
        )
    }
}

File: java-to-go-projects/05-port-scanner/go/scanner/result.go
package scanner

import "fmt"

type ScanResult struct {
	Host      string
	Port      int
	Open      bool
	LatencyMs int64
}

func (r ScanResult) String() string {
	if r.Open {
		return fmt.Sprintf("%-5d OPEN  (%dms)", r.Port, r.LatencyMs)
	}
	return fmt.Sprintf("%-5d closed", r.Port)
}
